### ISSUE TITLE
File with CR-LF line break will loaded error on Windows.

### ISSUE THREAD

**Issue by NEUGWB** [marcobambini/gravity#350]
According to [msdn](https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/read?view=msvc-160) __read_ a file with CR-LF line break will cause read size smaller than the file size.

**Comment by marcobambini** [comment]
Thanks @NEUGWB. Do you have a suggestion for a fix?

**Comment by NEUGWB** [comment]
> Thanks @NEUGWB. Do you have a suggestion for a fix?

I think use fopen instead of open, or an fdeof function may help.
```
bool fdeof(int fd)
{
  char c;
  int rdsize = read(fd, &c, 1);
  return rdsize == 0;
}
```
I don't know if there is a better way. :)

**Comment by wtjcn** [comment]
my solution, tested ok :

char *file_read(const char *path, size_t *len) {
    
    fd = open(path, O_RDONLY | O_BINARY);
                                              ~~~~~~~~~

**Comment by tDwtp** [comment]
Fixed with #365 
Had the same problem. Could have just read this instead, would have been a lot easier.
